### DISCUSSION TITLE
NumbaDeprecationWarning was given when running whisper

### DISCUSSION THREAD

**Discussion by LaoshuBaby** [openai/whisper#1344] [Show and tell]
When I run whisper, I was given such an error message. Is this related to that I can only use FP32? Or, will this have any negative effects on me? 

```
PS C:\Users\laoshubaby\Downloads> whisper python_hunter_ep40.mp3 --language Chinese
C:\Environment\Python\Python311\Lib\site-packages\whisper\timing.py:57: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @numba.jit
```

**Comment by omkar-334** [comment]
Go to timing.py, line 57
change `@numba.jit` to `@numba.jit(nopython=True)`
Hope this resolves the issue.

**Reply by LaoshuBaby** [reply]
> Go to timing.py, line 57 

However, it is generally not recommended to manually change the installed library functions? Maybe it's worth an issue and wait for an upstream developer to fix it instead of me changing in `/site-package`?

**Reply by omkar-334** [reply]
You should wait for the next major release.
This is just a temporary fix.

**Comment by Tex2002ans** [comment]
This numba `nopython` warning was already fixed in this PR on May 5th:

- https://github.com/openai/whisper/pull/1233

This fix isn't in the latest main Whisper release (v20230314), so if you want it, you'll have to:

- download the master commit

OR

- wait until next major release.

> However, it is generally not recommended to manually change the installed library functions?

All omkar-334 gave was the same exact line/patch that was fixed above.
